addons/hr_recruitment/models/hr_applicant.py

```diff
@@ -317,20 +317,17 @@ def _inverse_partner_email(self):
         for applicant in self:
             if not applicant.email_from:
                 continue
+            if not applicant.partner_name:
```

Comment:
this change the behavior a little, before if we had and a partner_id, we would not update the partner_name and not having it would not cause an error